lib/Agrammon/Model.pm6

```diff
@@ -446,10 +447,17 @@ class Agrammon::Model {
     }
 
     method output-print(Str $module, Str $output) {
-        %!output-print-cache ||= @!evaluation-order.map({
+        %!output-print-str-cache ||= @!evaluation-order.map({
             .taxonomy => %(.output.map({ .name => .print }))
         });
-        return %!output-print-cache{$module}{$output} // ''
+        return %!output-print-str-cache{$module}{$output} // ''
+    }
+
+    method should-print(Str $module, Str $output, @print-set) {
+        %!output-print-cache ||= @!evaluation-order.map({
+            .taxonomy => %(.output.map({ .name => (.split(',').List with .print) }))
+        });
+        !@print-set or ( %!output-print-cache{$module}{$output} // () ) ∩ @print-set
```

Comment:
I'm wondering if we ever have a case where we won't have an entry in the hash now? Unlike with the string case, we'll store `Empty` under outputs that have no prints, so the `// ()` is probably redundant.

Reply:
Removed, still producing the right output.